### PR TITLE
Fix remaining Neural Layer Tests

### PR DESCRIPTION
**Pull request by saatvikshah1994**
1. Replaced redundancies using fixture class and its helper methods.
2. Replaced manual for loops with shogun::linalg calls wherever feasible.

### CODE REVIEW

File: tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc [hunk 1]
```diff
@@ -30,75 +30,61 @@
  *
  * Written (W) 2015 Sanuj Sharma
  */
-
```
Comment (karlnapf) [inline tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc]: add your name if you like

**Comment by saatvikshah1994** [comment]
Yep no memory errors here.

File: tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc (hunk 1 above)
Comment (saatvikshah1994) [reply tests/unit/neuralnets/NeuralLeakyRectifiedLinearLayer_unittest.cc]: its fine - i didnt really add any logic just removed boilerplate

**Comment by karlnapf** [comment]
As usual, let’s watch out for the ci abd buildbot

**Pull request merged**